demos/python_demos/3d_segmentation_demo/3d_segmentation_demo.py

```diff
@@ -153,21 +171,35 @@ def resample_np(data, output_shape, order):
     return interpolation.zoom(data, zoom=factor, order=order)
 
 
-def read_image(test_data_path, series_name, sizes=(128, 128, 128)):
+def read_image(test_data_path, data_name, sizes=(128, 128, 128), is_series=True):
     images_list = []
     handle = None
     bboxes = np.zeros(shape=(len(DATA_SUFFIXES),) + (2, 3))
 
-    for j, s in enumerate(DATA_SUFFIXES):
-        image_handle = read_nii_header(test_data_path, series_name, s, separate_folder=False)
+    if is_series:
+        for j, s in enumerate(DATA_SUFFIXES):
+            image_handle = read_nii_header(test_data_path, data_name, s, separate_folder=False)
+            handle = image_handle
+            image = image_handle.get_data().astype(np.float32)
+
+            mask = image > 0.
+            bboxes[j] = bbox3(mask)
+            image = normalize(image, mask)
+
+            images_list.append(image.reshape((1, 1,) + image.shape))
+            original_shape = image.shape
+    else:
+        image_handle = read_nii_header(test_data_path, data_name, separate_folder=False)
         handle = image_handle
         image = image_handle.get_data().astype(np.float32)
 
-        mask = image > 0.
-        bboxes[j] = bbox3(mask)
-        image = normalize(image, mask)
-
-        images_list.append(image.reshape((1, 1,) + image.shape))
+        for j in (1, 3, 0, 2):
```

Comment:
Since order of reading different scans is crucial, and scans order, packed in single file, can  differs, may be add argument which will set the order:

3d_segmentation_demo.py ... --mri_sequence 1,3,0,2 ...

It may be not used at all, in fact from different data source could come completely different format of scans.

What do you think?

Reply:
`1, 3, 0, 2` is specified for data from [http://medicaldecathlon.com/](http://medicaldecathlon.com/)

Reply:
The answer could depend on existence of know data sets that use different sequence. Or maybe there is commonly used tool that generates only that sequence.
On the other hand, I don't think decision on adding key is really important.

My suggestion is not to introduce a new key until we will know there is a data set in a different format.

Reply:
Agree.